internal/nix/shell.go

```diff
@@ -129,9 +129,11 @@ func WithHistoryFile(historyFile string) ShellOption {
 
 // TODO: Consider removing this once plugins add env vars directly to binaries
 // via wrapper scripts.
-func WithEnvVariables(envVariables []string) ShellOption {
+func WithEnvVariables(envVariables map[string]string) ShellOption {
 	return func(s *Shell) {
-		s.env = append(s.env, envVariables...)
+		for k, v := range envVariables {
+			s.env = append(s.env, fmt.Sprintf("%s=%s", k, v))
+		}
```

Comment:
Can we have a utility function for this. Because we need both formats. I agree default form should be `map[string]string` but also have a converter from map[string]string to []string with `key=value` format can make things very easy.

Reply:
Agreed